File: KotlinMobileApp/Java/Start/app/src/main/java/com/rachelleboyette/catquiz/dtos/CatFactDto.java
package com.rachelleboyette.catquiz.dtos;

import com.rachelleboyette.catquiz.models.CatFact;

public class CatFactDto {
    private String fact;

    public CatFactDto(String fact) {
        this.fact = fact;
    }

    public static CatFact fromDto(CatFactDto dto) {
        return new CatFact(dto.fact);
    }
}

File: KotlinMobileApp/KotlinMobileApp/app/src/main/java/com/neudesic/catquiz/dtos/CatFactDto.kt
package com.neudesic.catquiz.dtos

import com.neudesic.catquiz.models.CatFact

class CatFactDto(val fact: String) {
    companion object Dto{
        fun fromDto(catDtos: List<CatFactDto>): List<CatFact>?{
            return null
        }

        fun fromDto(catDto: CatFactDto?) : CatFact? {
            return CatFact(catDto?.fact)
        }
    }

}